modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -478,12 +480,19 @@ private void exec() {
                     break;
                 case InstructionCodes.CALL:
                     cpIndex = operands[0];
-                    funcRefCPEntry = (FunctionRefCPEntry) constPool[cpIndex];
-                    functionInfo = funcRefCPEntry.getFunctionInfo();
+                    
+                    ConstantPoolEntry cpEntry = constPool[cpIndex];
+                    CallableUnitInfo callableUnitInfo;
+                    if (cpEntry.getEntryType() == EntryType.CP_ENTRY_FUNCTION_REF) {
```

Comment:
This approach is not good.

Can we have another instruction to invoke a transformer

Reply:
yes, will add